workflow/src/main/java/com/google/android/fhir/workflow/activity/ActivityFlow.kt

```diff
@@ -0,0 +1,351 @@
+ *
+ *     resumedOrderFlow.startPerform(draftEvent).onSuccess { performPhase ->
+ *       val updatedEvent =
+ *         performPhase.getEvent().copy().apply {
+ *           setStatus(EventStatus.INPROGRESS)
+ *           update { addNote(Annotation(MarkdownType("Event looks OK."))) }
+ *         }
+ *
+ *       performPhase
+ *         .update(updatedEvent)
+ *         .onSuccess { cachedRequestId = updatedEvent.logicalId }
+ *         .onFailure { fail("Should have succeeded", it) }
+ *     }
+ *   }
+ *   .onFailure { fail("Unexpected", it) }
+ *
+ *   val eventToResume =
+ *     repository
+ *       .read(MedicationDispense::class.java, IdType("MedicationDispense", cachedRequestId))
+ *       .let { CPGMedicationDispenseEvent(it) }
+ *
+ *   val resumedPerformFlow = ActivityFlow.of(repository, eventToResume)
+ *
+ *   check(resumedPerformFlow.getCurrentPhase() is Phase.EventPhase<*>) {
+ *     "Flow is in ${resumedPerformFlow.getCurrentPhase().getPhaseName()} "
+ *   }
+ *
+ *   (resumedPerformFlow.getCurrentPhase() as Phase.EventPhase<*>).complete()
+ * ```
+ */
+@Suppress(
+  "UnstableApiUsage", /* Repository is marked @Beta */
+)
+class ActivityFlow<R : CPGRequestResource<*>, E : CPGEventResource<*>>
+private constructor(
+  private val repository: Repository,
+  requestResource: R? = null,
+  eventResource: E? = null,
+) {
+
+  private var currentPhase: Phase
+
+  init {
+    currentPhase =
+      if (eventResource != null) {
+        PerformPhase(repository, eventResource)
+      } else if (requestResource != null) {
+        when (requestResource.getIntent()) {
+          Intent.PROPOSAL -> ProposalPhase(repository, requestResource)
+          Intent.PLAN -> PlanPhase(repository, requestResource)
+          Intent.ORDER -> OrderPhase(repository, requestResource)
+          else ->
+            throw IllegalArgumentException(
+              "Couldn't create the flow for ${requestResource.getIntent()} intent. Supported intents are 'proposal', 'plan' and 'order'.",
+            )
+        }
+      } else {
+        throw IllegalArgumentException(
+          "Either Request or Event is required to create a flow. Both can't be null.",
+        )
+      }
+  }
+
+  /**
+   * Returns the current phase of the flow. The users may check the type of flow by calling
+   * [Phase.getPhaseName] on the [getCurrentPhase] and then cast it to appropriate classes.
+   *
+   * The table below shows the mapping between the [PhaseName] and [Phase] implementations.
+   *
+   * | [PhaseName] | [Class]         |
+   * |-------------|-----------------|
+   * | [PROPOSAL]  | [ProposalPhase] |
+   * | [PLAN]      | [PlanPhase]     |
+   * | [ORDER]     | [OrderPhase]    |
+   * | [PERFORM]   | [PerformPhase]  |
+   */
+  fun getCurrentPhase(): Phase {
+    return currentPhase
+  }
+
+  /**
+   * Creates a draft plan resource based on the state of the [currentPhase].
+   *
+   * @return [R] if the action is successful, error otherwise.
```

Comment:
Correct the return kdoc.
```suggestion
   * @return [Result]<R> containing the draft plan resource if the action is successful, [Result.failure] otherwise.
```

Reply:
Done